frame/support/src/traits/metadata.rs

```diff
@@ -48,6 +48,10 @@ pub struct CallMetadata {
 	pub function_name: &'static str,
 	/// Name of the pallet to which the function belongs.
 	pub pallet_name: &'static str,
+	/// Index of the containing pallet specified in `construct_runtime`.
+	pub index: u8,
+	/// Version of the crate containing the pallet to which the function belongs.
+	pub crate_version: CrateVersion,
```

Comment:
IMO this doesn't make any sense here.

`CallMetadata` is about an instance of a call. Now you adding `crate_version` and the `index` which only works if this is an outer call.

`index` and `crate_version` should be moved above to `PalletInfoAccess`.

Reply:
That's what I thought too, I wasn't sure where it would make the most sense, but I'll try putting it in `PalletInfoAccess`. Thanks!

Reply:
Ty! :)